src/cloudsc_python/src/cloudsc4py/utils/validation.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+from __future__ import annotations
+import numpy as np
+from typing import TYPE_CHECKING
+
+from cloudsc4py.utils.numpyx import to_numpy
+
+if TYPE_CHECKING:
+    from typing import Tuple
+
+    from sympl._core.data_array import DataArray
+    from sympl._core.typingx import DataArrayDict
+
+    from cloudsc4py.utils.typingx import Storage
+
+
+def validate_storage_2d(src: Storage, trg: Storage) -> bool:
+    src_np = to_numpy(src)
+    trg_np = to_numpy(trg)
+    mi = min(src_np.shape[0], trg_np.shape[0])
+    mj = min(src_np.shape[1], trg_np.shape[1])
+    return np.allclose(src_np[:mi, :mj], trg_np[:mi, :mj])
+
+
+def validate_storage_3d(src: Storage, trg: Storage) -> bool:
+    src_np = to_numpy(src)
+    trg_np = to_numpy(trg)
+    mi = min(src_np.shape[0], trg_np.shape[0])
+    mj = min(src_np.shape[1], trg_np.shape[1])
+    mk = min(src_np.shape[2], trg_np.shape[2])
+    return np.allclose(src_np[:mi, :mj, :mk], trg_np[:mi, :mj, :mk])
```

Comment:
This is a bit of a problem, as the default values for `np.allclose(...)` are very low compared to what we usually do. Admittedly, non-bit-identical validation is currently still based on manually assessing the deviations, but the default values seem too lose (some value are more likely to diverge than others :grimacing: ). In any case, I'm happy to leave this as is for now, and improve the validation mechanism in a follow-up PR.

Reply:
By default, the absolute and relative tolerance are 1e-6 and 1e-5. I've checked that the validation succeeds up to an absolute and relative tolerance of 1e-18 and 1e-15. This could be closer to your validation process. I'm adapting the code to make the comparisons stricter.